ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -383,51 +383,50 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
       .asVectorOf(partyEntryParser)(connection)
   }
 
-  private val SQL_SELECT_MULTIPLE_PARTIES =
-    SQL(
-      "select parties.party, parties.display_name, parties.ledger_offset, parties.explicit, parties.is_local from parties, parameters where party in ({parties}) and parties.ledger_offset <= parameters.ledger_end"
-    )
-
-  private case class ParsedPartyData(
-      party: String,
-      displayName: Option[String],
-      ledgerOffset: Offset,
-      explicit: Boolean,
-      isLocal: Boolean,
-  )
-
-  private val PartyDataParser: RowParser[ParsedPartyData] = {
-    import com.daml.platform.store.Conversions.columnToOffset
+  val partyDetailsParser: RowParser[PartyDetails] = {
     import com.daml.platform.store.Conversions.bigDecimalColumnToBoolean
-    Macro.parser[ParsedPartyData](
-      "party",
-      "display_name",
-      "ledger_offset",
-      "explicit",
-      "is_local",
-    )
+    str("party") ~
+      str("display_name").? ~
+      bool("is_local") map { case party ~ displayName ~ isLocal =>
+        PartyDetails(
+          party = Ref.Party.assertFromString(party),
+          displayName = displayName,
+          isLocal = isLocal,
+        )
+      }
   }
 
-  private def constructPartyDetails(data: ParsedPartyData): PartyDetails =
-    PartyDetails(Ref.Party.assertFromString(data.party), data.displayName, data.isLocal)
-
-  def parties(parties: Seq[Ref.Party])(connection: Connection): List[PartyDetails] = {
-    import com.daml.platform.store.Conversions.partyToStatement
-    SQL_SELECT_MULTIPLE_PARTIES
-      .on("parties" -> parties)
-      .as(PartyDataParser.*)(connection)
-      .map(constructPartyDetails)
+  private def queryParties(parties: Set[String], connection: Connection): Vector[PartyDetails] = {
+    val partyFilter =
+      if (parties.isEmpty) cSQL""
+      else cSQL"party_entries.party in ($parties) AND"
+    SQL"""
+        SELECT
+          parties.party,
+          parties.display_name,
+          parties.is_local
+        FROM (
+          SELECT
+            party_entries.party party,
+            party_entries.display_name display_name,
+            party_entries.is_local is_local,
```

Comment:
Important: As we are retrieving `is_local` here and that one has a set-semantics, we must use `bool_or(party_entries.is_local)` to return the correct information wrt locality.

Reply:
ouch, complete misunderstanding from my part: I was thinking we should retrieve the latest.
That complicates query big time in complexity and performance.
Not sure there is a bool_or for Oracle, probably we need something more portable.

Reply:
as discussed offline with @meiersi-da , we switched to group-by aggregated table selects, and proper implementation for is_local